packages/teams-js/src/internal/communication.ts

```diff
@@ -305,6 +312,70 @@ function processMessage(evt: DOMMessageEvent): void {
 
 const shouldProcessMessageLogger = communicationLogger.extend('shouldProcessMessage');
 
+function sendValidateDomainRequest(messageOrigin: URL): Promise<boolean> {
+  const messageOriginUrl = sanitizeAndStrigifyURL(messageOrigin);
+  return sendAndUnwrap('validateDomains', messageOriginUrl);
+}
+
+export function validateOrigin(messageOrigin: URL): boolean {
+  // Check whether the url is in the pre-known allowlist or supplied by user
+  if (!isValidHttpsURL(messageOrigin)) {
+    validateOriginLogger(
+      'Origin %s is invalid because it is not using https protocol. Protocol being used: %s',
+      messageOrigin,
+      messageOrigin.protocol,
+    );
+    return false;
+  }
+  const messageOriginHost = messageOrigin.host;
+  let isVerified = false;
+  if (Communication.alreadyVerifiedDomains.has(messageOriginHost)) {
+    isVerified = Communication.alreadyVerifiedDomains.get(messageOriginHost);
+  } else {
+    sendValidateDomainRequest(messageOrigin).then((hubResponse: boolean) => {
+      isVerified = hubResponse;
+      Communication.alreadyVerifiedDomains.set(messageOriginHost, hubResponse);
+    });
+  }
+  if (isVerified) {
+    return true;
+  }
+  if (fallbackValidateOrigin(messageOriginHost)) {
+    return true;
+  }
+  if (validateAdditionalOrigins(messageOriginHost)) {
+    return true;
+  }
+  validateOriginLogger(
+    'Origin %s is invalid because it is not an origin approved by the host, this library or included in the call to app.initialize.\nOrigins approved by this library: %o\nOrigins included in app.initialize: %o',
+    messageOrigin,
+    validOrigins,
+    GlobalVars.additionalValidOrigins,
+  );
+  return false;
+}
+
+function isMessageComingFromParentOrigin(messageOrigin: string): boolean {
+  return (
+    Communication.parentOrigin &&
+    messageOrigin &&
+    messageOrigin === Communication.parentOrigin &&
+    Communication.alreadyVerifiedDomains.size === 0
```

Comment:
Why is this line needed?

Reply:
Whoops good catch, I was doing that for testing purposes, it is not needed. I'll remove it